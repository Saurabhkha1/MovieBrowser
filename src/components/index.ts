export * from './iconComponent';
export * from './header';
export * from './movieComponent';
export * from './loader';
