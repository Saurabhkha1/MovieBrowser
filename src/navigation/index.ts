export * from './bottomTabs';
export * from './stack';
