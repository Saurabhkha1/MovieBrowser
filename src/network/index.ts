import axios from 'axios';
import {BASE_URL, BEARER_TOKEN} from '../utils/constants';

type Movie = {
  id: number;
  title: string;
  poster_path: string;
  release_date: string;
  vote_average: number;
  overview: string;
};

export const fetchMovies = async (
  endpoint: string,
  page: number = 1,
): Promise<Movie[]> => {
  const response = await axios.get(`${BASE_URL}/movie/${endpoint}`, {
    params: {language: 'en-US', page},
    headers: {
      Authorization: `Bearer ${BEARER_TOKEN}`,
      Accept: 'application/json',
    },
  });
  return response.data.results;
};
