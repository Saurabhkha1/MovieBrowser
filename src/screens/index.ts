export * from './commonHome';
export * from './movieDetails';
