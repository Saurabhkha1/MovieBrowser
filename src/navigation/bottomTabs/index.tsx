import React from 'react';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import {CommonHome} from '../../screens';
import {IconComponent} from '../../components';
import {IconProp} from '@fortawesome/fontawesome-svg-core';
import {Icons} from '../../utils/icons';
import {Colors} from '../../utils/colors';

type BottomTabParamList = {
  NowPlaying: {title: string; icon: IconProp};
  Popular: {title: string; icon: IconProp};
  TopRated: {title: string; icon: IconProp};
  Upcoming: {title: string; icon: IconProp};
};

const Tab = createBottomTabNavigator<BottomTabParamList>();

export const BottomTabNavigator = () => {
  const getIcon = (icon: IconProp, size?: number, color?: string) => {
    return <IconComponent icon={icon} size={size} color={color} />;
  };

  return (
    <Tab.Navigator
      screenOptions={{
        headerShown: false,
        tabBarActiveTintColor: Colors.darkBlue,
      }}>
      <Tab.Screen
        name="NowPlaying"
        component={CommonHome}
        options={{
          tabBarIcon: ({focused}) =>
            getIcon(
              Icons.nowPlaying,
              25,
              focused ? Colors.darkBlue : Colors.black,
            ),
          tabBarLabel: 'Now Playing',
        }}
        initialParams={{title: 'Now Playing', icon: Icons.nowPlaying}}
      />
      <Tab.Screen
        name="Popular"
        component={CommonHome}
        options={{
          tabBarIcon: ({focused}) =>
            getIcon(
              Icons.popular,
              25,
              focused ? Colors.darkBlue : Colors.black,
            ),
          tabBarLabel: 'Popular',
        }}
        initialParams={{title: 'Popular', icon: Icons.popular}}
      />
      <Tab.Screen
        name="TopRated"
        component={CommonHome}
        options={{
          tabBarIcon: ({focused}) =>
            getIcon(
              Icons.topRated,
              25,
              focused ? Colors.darkBlue : Colors.black,
            ),
          tabBarLabel: 'Top Rated',
        }}
        initialParams={{title: 'Top Rated', icon: Icons.topRated}}
      />
      <Tab.Screen
        name="Upcoming"
        component={CommonHome}
        options={{
          tabBarIcon: ({focused}) =>
            getIcon(
              Icons.upcoming,
              25,
              focused ? Colors.darkBlue : Colors.black,
            ),
        }}
        initialParams={{title: 'Upcoming', icon: Icons.upcoming}}
      />
    </Tab.Navigator>
  );
};
