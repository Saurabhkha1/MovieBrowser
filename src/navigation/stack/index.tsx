import React from 'react';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import {BottomTabNavigator} from '../bottomTabs';
import {MovieDetails} from '../../screens';
import {Movie} from '../../redux/movieSlice';

export type RootStackParamList = {
  BottomTabs: undefined;
  MovieDetails: {movie: Movie};
};

const Stack = createNativeStackNavigator<RootStackParamList>();

export const StackNavigator = () => {
  return (
    <Stack.Navigator screenOptions={{headerShown: false}}>
      <Stack.Screen name="BottomTabs" component={BottomTabNavigator} />
      <Stack.Screen name="MovieDetails" component={MovieDetails} />
    </Stack.Navigator>
  );
};
