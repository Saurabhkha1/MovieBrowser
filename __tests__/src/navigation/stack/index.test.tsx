import React from 'react';
import {create} from 'react-test-renderer';
import {NavigationContainer} from '@react-navigation/native';
import {StackNavigator} from '../../../../src/navigation';

jest.mock('@react-navigation/native', () => {
  return {
    NavigationContainer: ({children}: any) => <>{children}</>,
  };
});

jest.mock('@react-navigation/native-stack', () => {
  return {
    createNativeStackNavigator: jest.fn().mockReturnValue({
      Navigator: ({children}: any) => <>{children}</>,
      Screen: ({component}: any) => {
        const Component = component;
        return <Component />;
      },
    }),
  };
});

jest.mock('../../../../src/navigation/bottomTabs', () => {
  return {
    BottomTabNavigator: () => <div>BottomTabNavigator Component</div>,
  };
});

jest.mock('../../../../src/screens', () => {
  return {
    MovieDetails: () => <div>MovieDetails Component</div>,
  };
});

describe('StackNavigator', () => {
  it('renders the BottomTabs screen', () => {
    const tree = create(
      <NavigationContainer>
        <StackNavigator />
      </NavigationContainer>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders the MovieDetails screen', () => {
    const tree = create(
      <NavigationContainer>
        <StackNavigator />
      </NavigationContainer>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
