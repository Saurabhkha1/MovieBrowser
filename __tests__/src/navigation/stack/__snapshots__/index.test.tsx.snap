// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackNavigator renders the BottomTabs screen 1`] = `
[
  <div>
    BottomTabNavigator Component
  </div>,
  <div>
    MovieDetails Component
  </div>,
]
`;

exports[`StackNavigator renders the MovieDetails screen 1`] = `
[
  <div>
    BottomTabNavigator Component
  </div>,
  <div>
    MovieDetails Component
  </div>,
]
`;
